python/graphstorm/distiller/distiller.py

```diff
@@ -0,0 +1,273 @@
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Distill wrapper.
+"""
+
+import torch as th
+from torch.nn.parallel import DistributedDataParallel
+import os
+import json
+import logging
+
+import graphstorm as gs
+from .utils import to_device
+from ..utils import barrier
+
+class GSdistiller():
+    """ GNN distiller.
+    This class defines the running pipeline of distillation.
+    It trains the student model and saves the checkpoint.
+
+    Parameters
+    ----------
+    model : GSDistilledModel
+        The student model.
+    """
+    def __init__(self, model):
+        self.rank = gs.get_rank()
+        self.model = model
+        self._device = None
+
+    def fit(
+        self, 
+        train_data_mgr,
+        eval_data_mgr,
+        distill_lr,
+        saved_path,
+        save_model_frequency,
+        eval_frequency,
+        max_distill_step,
+        on_cpu=False,
+    ):
+        """ Distill function, which trains student model.
+
+        Parameters
+        ----------
+        train_data_mgr : DataManager
+            Training data manager.
+        eval_data_mgr : DataManager
+            Evaluation data manager.
+        distill_lr : float
+            Learning rate for distillation.
+        saved_path : str
+            Path to save the model.
+        save_model_frequency : int,
+            Interval to save the model checkpoint.
+        eval_frequency : int,
+            Interval for evaluation.
+        max_distill_step : int,
+            Maximum steps for distillation training.
+        on_cpu : bool
+            Whether the distillation will be conducted on cpu.
+        """
+
+        # TODO (HZ): add flexibility to specify different optimizers
+        optimizer = th.optim.Adam(self.model.parameters(), lr=distill_lr)
+        self.model.to(self.device)
+        model = DistributedDataParallel(self.model, device_ids=None if on_cpu else [self.device],
+                                        output_device=None if on_cpu else self.device)
+
+        index = 0
+        distill_step = 0
+        complete=False
+        while complete is False:
+            barrier()
+            logging.info(f"Train {index + 1}-th shard by trainer {self.rank}")
+            complete, distill_step = self.train_shard(
+                distill_step=distill_step,
+                model=model, 
+                optimizer=optimizer, 
+                train_data_mgr=train_data_mgr, 
+                eval_data_mgr=eval_data_mgr,
+                saved_path=saved_path,
+                save_model_frequency=save_model_frequency,
+                eval_frequency=eval_frequency,
+                max_distill_step=max_distill_step,
+            )
+            index += 1
+
+    def save_student_model(self, model, saved_path, distill_step):
```

Comment:
i thought we agreed to move this function to the model class.

Reply:
Trainer's save model will finally call model's save model.
But here, you directly get parameter_dicts from model to save.

Reply:
You should call 
``model.module.save_model(checkpoint_path)`` in L112.
And move the codes L112-L120 to GSDistilledModel.

Otherwise, people can not implement their own DistilledModel.

Reply:
addressed by new commit.